Cargo.toml

```
@@ -22,23 +22,27 @@ ark-poly = { git = "https://github.com/arkworks-rs/algebra", default-features =
 ark-std = { git = "https://github.com/arkworks-rs/utils", default-features = false }
 bench-utils = { git = "https://github.com/arkworks-rs/utils" }
 ark-relations = { git = "https://github.com/arkworks-rs/snark", default-features = false }
-ark-snark = { git = "https://github.com/arkworks-rs/snark", default-features = false }
+ark-crypto-primitives = { git = "https://github.com/arkworks-rs/crypto-primitives.git", branch = "main", default-features = false,  features = [ "r1cs" ] }
```

We should enable the `r1cs` feature only when we use it.

```suggestion
ark-crypto-primitives = { git = "https://github.com/arkworks-rs/crypto-primitives.git", branch = "main", default-features = false }
```